src/main/java/io/managed/services/test/client/BaseApi.java

```diff
@@ -1,111 +1,48 @@
 package io.managed.services.test.client;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
 import io.managed.services.test.Environment;
 import io.managed.services.test.RetryUtils;
 import io.managed.services.test.ThrowingSupplier;
 import io.managed.services.test.ThrowingVoid;
 import io.managed.services.test.client.exception.ApiGenericException;
-import io.managed.services.test.client.exception.ApiUnauthorizedException;
-import io.managed.services.test.client.exception.ApiUnknownException;
 import lombok.extern.log4j.Log4j2;
-import okhttp3.FormBody;
-import okhttp3.OkHttpClient;
-import okhttp3.Request;
-
-import java.util.concurrent.atomic.AtomicReference;
 
 @Log4j2
 public abstract class BaseApi {
 
     private String defaultSSOUrl = Environment.REDHAT_SSO_URI + "/auth/realms/" + Environment.REDHAT_SSO_REALM + "/protocol/openid-connect/token";
 
-    private final String offlineToken;
     private final String url;
 
-    protected BaseApi(String offlineToken) {
-        this.offlineToken = offlineToken;
+    protected BaseApi() {
         this.url = defaultSSOUrl;
     }
 
-    protected BaseApi(String offlineToken, String url) {
-        this.offlineToken = offlineToken;
-        this.url = url;
-    }
-
     /**
      * @param e Exception
-     * @return ApiUnknownException | null if the passed Exception can't be converted
+     * @return ApiGenericException | null if the passed Exception can't be converted
      */
-    protected abstract ApiUnknownException toApiException(Exception e);
-
-    protected abstract void setAccessToken(String t);
+    protected abstract ApiGenericException toApiException(Exception e);
 
     private <A> A handleException(ThrowingSupplier<A, Exception> f) throws ApiGenericException {
         try {
             return f.get();
+        }  catch (ApiGenericException e) {
+            throw e;
         } catch (Exception e) {
```

Comment:
based on failed test: 
basically anywhere here we need to call 
public static ApiGenericException apiException(ApiUnknownException e) 
located in ApiGenericException  class. 

THe reason is that otherwise GenericException will never be cast correctly to more specific kind